controllers/chat/initChatFromPost.js

```diff
@@ -0,0 +1,68 @@
+const {User} = require('../../databases/models');
+const {StreamChat} = require('stream-chat');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+const {responseSuccess} = require('../../utils/Responses');
+const {CHANNEL_TYPE} = require('../../helpers/constants');
+const UsersFunction = require('../../databases/functions/users');
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const initChatFromPost = async (req, res) => {
+  const {postId, targetUserId} = req.body;
+
+  const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  try {
+    const [userModel, targetUserModel] = await Promise.all([
+      UsersFunction.findUserById(User, req?.userId),
+      UsersFunction.findUserById(User, targetUserId)
+    ]);
+    /**
+     * @type {import('stream-chat').OwnUserResponse}
+     */
+    const user = {
+      name: userModel?.username,
+      id: req.userId,
+      image: userModel?.profile_pic_path,
+      username: userModel?.username
+    };
+
+    await client.connectUser(user, req.token);
+
+    const channel = client.channel('messaging', `fromPost-${postId}`);
```

Comment:
@brm-stnd if channel id only includes postId, then how can we differentiate chat initiated by user A or user B?
I think we should have another differentiation.
e.g. `fromPost-${postId}-${selfUserId}`

This way we can know that this is chat initiated by selfUserId and at the postId